test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -22,18 +22,13 @@
 
 #define MAX_NUM_IFACES           2
 #define MAX_TM_SYSTEMS           3
-#define NUM_LEVELS               3
```

Comment:
I have not yet properly reviewed this last patch (add five level test suite). Since the patch seems unrelated to the API changes in this PR, you might be able to get the API changes merged quicker if this patch patch was moved to a separate PR so that the API change related patches would not have to wait until it is ready.

Could you also split the refactoring and the actual addition of new tests to separate patches. That would make it easier to see what code change is related to what and it would help in case later someone has to investigate a problem related either to the new test cases or to the refactoring.